add goal tolerance examples

gautz opened the pull request:
### Description

- examples related to https://github.com/ros-planning/moveit/pull/2780

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

jackcenter commented on doc/move_group_interface/move_group_interface_tutorial.rst:
```diff
@@ -38,8 +38,10 @@ See the `YouTube video <https://youtu.be/_5siHkFQPBQ>`_ at the top of this tutor
+ 8. The robot moves its arm to a pose goal with orientation tolerance, avoiding collision with the box.
+ 9. The robot moves its arm to a pose goal with position tolerance, avoiding collision with the box.
+ 10. The object is detached from the wrist (its color will change back to green).
+ 11. The object is removed from the environment.
```
If there is a video of the new demo, it would be great to update the youtube video demo link to show this new full process.

gautz replied:
could you suggest an updated tutorial steps numbered list if you see some improvement possible?

jackcenter replied:
I made a suggestion below to move your new steps after step 6.

gautz replied:
I chose to detach the object before planning with goal tolerance. Then the motion with and without object attached is kept one after the other.

jackcenter commented on doc/move_group_interface/src/move_group_interface_tutorial.cpp:
```diff
@@ -452,6 +452,56 @@ int main(int argc, char** argv)
+  move_group_interface.setPoseTarget(pose_above_collision_object);
+
+  std::vector<double> goal_orientation_tolerance_xyz = { 1e-3, M_PI / 2, M_PI / 4 };
+  move_group_interface.setGoalOrientationToleranceXYZ(goal_orientation_tolerance_xyz);
```
I wasn't able to get this to build due to the "setGoalOrientationToleranceXYZ" method not being in the move_group_interface. Is this new functionality that also needs to be added?

jackcenter commented:
During the new demos, where does the object being gripped go? It looks like it doesn't get accounted for in the simulation I ran.

https://user-images.githubusercontent.com/23001861/127402130-aed46ad7-f31a-48b7-adae-947fc2e4c1bc.mp4

gautz commented:
indeed the motion with goal tolerance is realized without object attached. Is this an issue?

jackcenter commented:
> During the new demos, where does the object being gripped go? It looks like it doesn't get accounted for in the simulation I ran.
> 
>  goalToleranceDemo.mp4

jackcenter commented:
Accidently closed this, it's open again.

In regard to the last comment, it's not really an issue so much as I wanted to make sure it's what you intended. Since the object is attached at this point (starting at step 7) would it make sense to plan the object attached? Maybe the quickest fix is just to reorder the steps and put your new demos right after 6 instead of after 7.

gautz commented:
I changed the demo and updated the corresponding text to follow the video attached

https://user-images.githubusercontent.com/64833674/129591240-43e2c11e-9169-47dc-a221-cc58842423cd.mp4

jackcenter commented:
Cool video. Looks good to me. If you post this video up on YouTube and link it to the tutorial page it would be a nice addition! But it looks like everything works and flows nicely.

jackcenter reviewed (approved):
Everything built and ran correctly. Relates to  https://github.com/ros-planning/moveit/pull/2780

gautz commented:
I posted the video on Youtube https://youtu.be/xwB7tpZK9-o

jackcenter commented:
Thanks for pinging me on this and sorry it has taken so long for a review. I'll take a look today or early next week and make sure this and the other PR moves forward.

jackcenter commented on doc/move_group_interface/move_group_interface_tutorial.rst:
```diff
@@ -28,18 +28,21 @@ After a short moment, the RViz window should appear and look similar to the one
 Expected Output
 ---------------
-See the `YouTube video <https://youtu.be/_5siHkFQPBQ>`_ at the top of this tutorial for expected output. In RViz, we should be able to see the following:
+See the `YouTube video <https://youtu.be/xwB7tpZK9-o>`_ at the top of this tutorial for expected output. In RViz, we should be able to see the following:
```
New video shows the new functionality and looks good. Similar to the old video.

jackcenter commented on doc/move_group_interface/src/move_group_interface_tutorial.cpp:
```diff
@@ -467,6 +468,58 @@ int main(int argc, char** argv)
   visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window once the new object is detached from the robot");
 
+  // Moving to a goal pose with orientation tolerances
+  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Getting some "WARNING: Title underline too short" warnings here when building this locally. But the site looks okay.

gautz replied:
solved

jackcenter commented on doc/move_group_interface/src/move_group_interface_tutorial.cpp:
```diff
@@ -467,6 +468,58 @@ int main(int argc, char** argv)
+  visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window once the plan is complete");
+
+  // Moving to a goal pose with position tolerances
+  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Getting some "WARNING: Title underline too short" warnings here when building this locally. But the site looks okay.

gautz replied:
solved

jackcenter reviewed (approved):
Depends on [moveit #2780](https://github.com/ros-planning/moveit/pull/2780). That will need to be merged before this can be done.
